client/src/store/slices/tourSlice/tourSlice.js

```diff
@@ -19,16 +19,10 @@ const tourSlice = createSlice({
 
 export const { setTour } = tourSlice.actions;
 
-export const fetchTour = (tourId) => async (dispatch) => {
-  try {
-    const { data, status } = await axios(`/products/${tourId}`);
+export const fetchTour = (itemNo) => async (dispatch) => {
+  const data = await getProduct(itemNo);
 
-    if (status) {
-      dispatch(setTour(data));
-    }
-  } catch (err) {
-    console.error(err.message);
-  }
+  dispatch(setTour(data));
 };
```

Comment:
you will use it like extraReducer and createAsyncThunk

the same advice for other async functions in your slices

Reply:
i will decide this in the refactoring phase, if createAsyncThunk will make it simpler and more readable then i will go for it, if in the case of this slice it will be extra uneeded code then i wont.